server/sonar-server/src/test/java/org/sonar/server/rule/index/RuleIndexTest.java

```diff
@@ -595,115 +570,278 @@ public void search_by_profile_and_active_severity() {
 
   @Test
   public void all_tags() {
-    indexRules(
-      newDoc(RuleKey.of("java", "S001")).setAllTags(asList("tag1", "sys1", "sys2")),
-      newDoc(RuleKey.of("java", "S002")).setAllTags(asList("tag2")));
+    OrganizationDto organization = dbTester.organizations().insert();
 
-    assertThat(index.terms(RuleIndexDefinition.FIELD_RULE_ALL_TAGS, null, 10)).containsOnly("tag1", "tag2", "sys1", "sys2");
+    RuleDefinitionDto rule1 = createRule(setSystemTags("sys1", "sys2"));
+    createRuleMetadata(rule1, organization,
+      setOrganizationUuid(organization.getUuid()),
+      setTags("tag1"));
+
+    RuleDefinitionDto rule2 = createRule(setSystemTags());
+    createRuleMetadata(rule2, organization,
+      setOrganizationUuid(organization.getUuid()),
+      setTags("tag2"));
+
+    assertThat(index.listTags(organization, null, 10)).containsOnly("tag1", "tag2", "sys1", "sys2");
+  }
+
+  @Test
+  public void all_tags_minds_the_oranization() {
```

Comment:
to be exact, a key that does not relate to an existing organization, for example "missing".